internal/cmd/request.go

```diff
@@ -105,13 +107,107 @@ func PrintCMDResponseFull(request any, response core.APIResponse, responseType a
 		fmt.Printf("\nAutograder Response:\n---\n%s\n---\n", util.MustToJSONIndent(response))
 	}
 
+	successfulConversion := false
+	customPrintOutput := ""
 	if customPrintFunc != nil {
-		fmt.Println(customPrintFunc(response))
-	} else if responseType == nil {
-		fmt.Println(util.MustToJSONIndent(response.Content))
+		customPrintOutput, successfulConversion = customPrintFunc(response)
+	}
+
+	output := ""
+	if successfulConversion {
```

Comment:
If you are going to use a nested if, then I don't think you need this case.
Just set `output` on the custom print call and only enter this conditional if the on `!successfulConversion`.